### PR TITLE
binding config schema validation for unused properties

### LINKED ISSUE #808: Report unused properties based on binding definition
**Describe the desired outcome from the user's perspective**
As a user editing a zilla config, I would like to get validation feedback for any valid property definitions that are not used or invalid based on other parts of the config. It is difficult to self diagnose issues when the config is visibly valid but the functionality isn't. 

**Acceptance criteria**
 - When a `kind` of binding determines which properties are needed or ignored, validate the properties and report to the user.

**Additional context**
One example is between the `cache_server` and `cache_client` kinds of `kafka` binding. `cache_server` can have `bootstrap` topics and `cache_client` does not need it.

### PR DESCRIPTION
**Pull request by aDaemonThread**
Fixes #808

### CODE REVIEW

**Review by jfallows — DISMISSED** [review]

File: specs/binding-http.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/schema/http.schema.patch.json [hunk 1]
```diff
@@ -388,6 +388,548 @@
...
+                                                }
+                                            ],
+                                            "additionalProperties": false
+                                        }
+                                    }
+                                },
+                                "additionalProperties": false
+                            }
+                        }
+                    }
```
Comment (jfallows) [inline specs/binding-http.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/schema/http.schema.patch.json:932]: Given that `server` options are a superset of `client` options, is there not a way to specify the common subset once, and then conditionally include the additional options properties for `server`?

**Review by jfallows — CHANGES_REQUESTED** [review]

**Comment by aDaemonThread** [comment]
Note:

Prior to this change, `options` & properties declared under options can be configured irrespective of the value configured for `kind`.
Below mentioned changes are done to restrict the configuration of unused properties. 

---------------------------
binding-grpc
if `kind: server`, options (services) properties are allowed 
if `kind: client`, options properties are not allowed

binding-kafka
if `kind: cache_client`, options (topics(name, key, value)) properties are allowed 
if `kind: cache_server`, options (bootstrap, topics(name, defaultOffset, defaultOffset, key, value)) properties are allowed 
if `kind: client`, options (servers, sasl) properties are allowed 

binding-mqtt
if `kind: server`, options (versions, authorization, topics) properties are allowed 
if `kind: client`, options properties are not allowed

binding-tls
if `kind: server`, options properties are allowed 
if `kind: client`, options properties are allowed 
if `kind: proxy`, options properties are not allowed

binding-ws
if `kind: server`, options properties are not allowed 
if `kind: client`, options(protocol, scheme, authority, path) properties are allowed

binding-http
if `kind: server`, options(versions, overrides, requests, access-control, authorization) properties are allowed 
if `kind: client`, options(versions, overrides, requests) properties are allowed 

---------------------------

**Review by jfallows — APPROVED** [review]

File: specs/binding-grpc.spec/src/main/scripts/io/aklivity/zilla/specs/binding/grpc/config/client.when.yaml [hunk 2]
```diff
@@ -19,9 +19,6 @@ bindings:
   app0:
     type: grpc
     kind: client
-    options:
-      services:
-        - protobuf/echo.proto
```
Comment (akrambek) [inline specs/binding-grpc.spec/src/main/scripts/io/aklivity/zilla/specs/binding/grpc/config/client.when.yaml:24]: why are we removing this?

File: specs/engine.spec/src/main/scripts/io/aklivity/zilla/specs/engine/config/server.binding.with.entry.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        - test.gauge
+        - test.histogram
+    exporters:
+        test0:
+            type: test
+bindings:
+    net0:
+        type: test
+        kind: server
+        entry: entry0
```
Comment (akrambek) [inline specs/engine.spec/src/main/scripts/io/aklivity/zilla/specs/engine/config/server.binding.with.entry.yaml:46]: I don't see any reference to `entry0` is this expected?

Comment (aDaemonThread) [reply specs/engine.spec/src/main/scripts/io/aklivity/zilla/specs/engine/config/server.binding.with.entry.yaml:46]: this script is added to test negative scenario(if kind: server, entry property is not allowed). expected output is `JsonException`. So I thought having an entry defined might not be required.

**Review by akrambek — APPROVED** [review]

**Review by bmaidics — APPROVED** [review]

**Review by attilakreiner — APPROVED** [review]

**Pull request merged**